core/src/components/datetime/datetime.tsx

```diff
@@ -1386,8 +1488,9 @@ export class Datetime implements ComponentInterface {
             ...findPart,
           });
 
+          // wheel pickers don't support multiple selection, so assume this won't be an array
           this.setActiveParts({
-            ...this.activeParts,
+            ...(this.activeParts as DatetimeParts),
```

Comment:
~ Typescript should be able to automatically narrow the type if we add checks that validate that `activeParts` is not an array. That would be my preference over type casting, which doesn't add functional safety if somehow it is set to an array in the implementation.